core/src/HIP/Kokkos_HIP_ParallelFor_MDRange.hpp

```diff
@@ -159,10 +159,7 @@ class ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>, HIP> {
     using closure_type =
         ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>, HIP>;
     unsigned block_size = hip_get_max_blocksize<closure_type, LaunchBounds>();
-    if (block_size == 0)
-      Kokkos::Impl::throw_runtime_exception(
-          std::string("Kokkos::Impl::ParallelFor< HIP > could not find a valid "
-                      "tile size."));
+    KOKKOS_ASSERT(block_size > 0);
```

Comment:
Is it correct that before this change, it would throw in both debug and release, and now it would silently do nothing in release mode ?

Reply:
Before we launch a kernel we check that the block size is not zero. If if is zero, we throw. So you still get an exception, it just happens later than now.

Reply:
I don't get your argument.  It is preferable to fail early and inject as much information as possible.
`KOKKOS_ABORT()` does not spell out file and line number, so we would be solely relying on the stack trace that gets printed.
I think that a special message describing what went wrong is better.

Reply:
Also if I read this correct it now says: if you are in release mode you get a throw you could catch, but in debug mode we unconditionally abort? That seems weird.